pkg/jobstore/context.go

```diff
@@ -0,0 +1,41 @@
+package jobstore
+
+import (
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+const maxTracingDuration = 10 * time.Millisecond
+
+// TracingContext is a context that can be used to trace the duration of a transaction
+// and log a debug message if it exceeds a certain threshold.
+type TracingContext struct {
+	TxContext
+	startTime time.Time
+}
+
+// NewTracingContext creates a new tracing context
+func NewTracingContext(ctx TxContext) *TracingContext {
+	return &TracingContext{
+		TxContext: ctx,
+		startTime: time.Now(),
+	}
+}
+
+func (t TracingContext) Commit() error {
+	if time.Since(t.startTime) > maxTracingDuration {
+		log.Debug().Msgf("transaction took longer than %s to commit", maxTracingDuration.String())
+	}
+	return t.TxContext.Commit()
+}
+
+func (t TracingContext) Rollback() error {
+	if time.Since(t.startTime) > maxTracingDuration {
+		log.Debug().Msgf("transaction took longer than %s to rollback", maxTracingDuration.String())
+	}
+	return t.TxContext.Rollback()
+}
```

Comment:
If I see these log lines, how will I know which transaction is the culprit? I think all I will see is:
`DEBUG transaction took longer than 10ms to rollback` which isn't explicit enough to be helpful.
Please remove this wrapper and instead opt for actionable log line in the job store on its respective methods.

Reply:
The log lines are useful in knowing something is wrong that requires investigation. They are not as great as metrics or actual traces, but this is a cheap and quick solution to call for investigation until we have better ways for tracing. These logs are specially useful now as we've just introduced multi-op transaction and can be removed after some time.

Log lines in in the job store methods are not useful as the they are not the ones holding the transaction. The `StateUpdaterPlanner` for example is a transaction holder and would be a bug if it is doing many slow operations while holding the transaction.

For now I've added the context in the log statement, and we can always add more logging contexts in the future to identify the source operation https://github.com/bacalhau-project/bacalhau/pull/3996/commits/33c2a51fd072486c37813f19388fa04b7793d07c